core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1976,6 +1976,56 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  /**
+   * Returns a qualified path object.
+   * @param defaultUri default uri corresponding to the filesystem provided.
+   * @param workingDir the working directory for the particular child path wd-relative names.
+   * @param path Path instance based on the path string specified by the user.
+   * @return Returns a qualified path object
+   */
+  def makeQualified(defaultUri: URI, workingDir: Path, path: Path): Path = {
+    val pathUri = {
+      if (!path.isAbsolute()) {
+        new Path(workingDir, path).toUri()
+      }
+      else {
```

Comment:
Nit: pull this up onto the previous line. I personally would suggest inverting the if condition for a little more clarity, while you're at it, but that's just a matter of taste.